From 1e6ef317f2676d19627a40fe2cd09b1e62934538 Mon Sep 17 00:00:00 2001
From: Ken Raffenetti <raffenet@mcs.anl.gov>
Date: Wed, 18 Oct 2017 14:39:34 -0500
Subject: [PATCH] config: Add embedded mode

Add a configure flag to support building UCX as a subproject of
another project. When configured for embedded mode, we avoid
installing any header files.
---
 configure.ac        | 5 +++++
 src/ucm/Makefile.am | 2 ++
 src/ucp/Makefile.am | 2 ++
 src/ucs/Makefile.am | 2 ++
 src/uct/Makefile.am | 2 ++
 5 files changed, 13 insertions(+)

diff --git a/configure.ac b/configure.ac
index 74284add..1a257d64 100644
--- a/configure.ac
+++ b/configure.ac
@@ -77,6 +77,11 @@ AC_C_RESTRICT
 m4_include([config/m4/pkg.m4])
 PKG_PROG_PKG_CONFIG
 
+AC_ARG_ENABLE([embedded],
+             [AS_HELP_STRING([--enable-embedded],
+                             [Enable embedded mode @<:@default=no@:>@])
+             ])
+AM_CONDITIONAL([EMBEDDED], [test x"$enable_embedded" = x"yes"])
 
 #
 # Save config flags for version dump tool
diff --git a/src/ucm/Makefile.am b/src/ucm/Makefile.am
index 2215ddd2..d5a59326 100644
--- a/src/ucm/Makefile.am
+++ b/src/ucm/Makefile.am
@@ -24,8 +24,10 @@ libucm_la_CFLAGS = \
 	$(BASE_CFLAGS) \
 	$(CFLAGS_NO_DEPRECATED)
 
+if !EMBEDDED
 nobase_dist_libucm_la_HEADERS = \
 	api/ucm.h
+endif
 
 noinst_HEADERS = \
 	event/event.h \
diff --git a/src/ucp/Makefile.am b/src/ucp/Makefile.am
index 5a2bb6a6..f96a3f48 100644
--- a/src/ucp/Makefile.am
+++ b/src/ucp/Makefile.am
@@ -13,11 +13,13 @@ libucp_la_LDFLAGS  = -ldl -version-info $(SOVERSION)
 libucp_la_LIBADD   = ../ucs/libucs.la ../uct/libuct.la
 libucp_ladir       = $(includedir)/ucp
 
+if !EMBEDDED
 nobase_dist_libucp_la_HEADERS = \
 	api/ucp_compat.h \
 	api/ucp_def.h \
 	api/ucp_version.h \
 	api/ucp.h
+endif
 
 noinst_HEADERS = \
 	amo/amo.inl \
diff --git a/src/ucs/Makefile.am b/src/ucs/Makefile.am
index de930eee..6b930964 100644
--- a/src/ucs/Makefile.am
+++ b/src/ucs/Makefile.am
@@ -17,6 +17,7 @@ libucs_la_LIBADD   = \
 	$(LIBM) \
 	$(top_builddir)/src/ucm/libucm.la
 
+if !EMBEDDED
 nobase_dist_libucs_la_HEADERS = \
 	algorithm/crc.h \
 	algorithm/qsort_r.h \
@@ -55,6 +56,7 @@ nobase_dist_libucs_la_HEADERS = \
 	type/status.h \
 	type/thread_mode.h \
 	type/cpu_set.h
+endif
 
 noinst_HEADERS = \
 	datastruct/arbiter.h \
diff --git a/src/uct/Makefile.am b/src/uct/Makefile.am
index 930f73b9..9a8f988e 100644
--- a/src/uct/Makefile.am
+++ b/src/uct/Makefile.am
@@ -15,11 +15,13 @@ libuct_la_LDFLAGS  = -ldl -version-info $(SOVERSION)
 libuct_la_LIBADD   = $(LIBM) ../ucs/libucs.la
 libuct_ladir       = $(includedir)/uct
 
+if !EMBEDDED
 nobase_dist_libuct_la_HEADERS = \
 	api/tl.h \
 	api/uct_def.h \
 	api/uct.h \
 	api/version.h
+endif
 
 noinst_HEADERS = \
 	base/addr.h \
-- 
2.11.0

